geomesa-raster/src/main/scala/org/locationtech/geomesa/raster/index/RasterEntry.scala

```diff
@@ -0,0 +1,120 @@
+import org.locationtech.geomesa.features.ScalaSimpleFeatureFactory
+import org.locationtech.geomesa.raster._
+import org.locationtech.geomesa.raster.data.Raster
+import org.opengis.feature.simple.SimpleFeature
+
+object RasterEntry {
+
+  val encoder = IndexValueEncoder(rasterSft, INTERNAL_GEOMESA_VERSION)
+
+  def encodeIndexCQMetadata(metadata: DecodedIndexValue): Array[Byte] = {
+    encodeIndexCQMetadata(metadata.id, metadata.geom, metadata.date)
+  }
+
+  def encodeIndexCQMetadata(uniqId: String, geometry: Geometry, dtg: Option[Date]): Array[Byte] = {
+    val metadata = ScalaSimpleFeatureFactory.buildFeature(rasterSft, List(geometry, dtg.orNull), uniqId)
+    encodeIndexCQMetadata(metadata)
+  }
+
+  def encodeIndexCQMetadata(sf: SimpleFeature): Array[Byte] = encoder.encode(sf)
+
+  def decodeIndexCQMetadata(k: Key): DecodedIndexValue = {
+    decodeIndexCQMetadata(k.getColumnQualifierData.toArray)
+  }
+
+  def decodeIndexCQMetadataToSf(cq: Array[Byte]): SimpleFeature = encoder.decode(cq)
+
+  def decodeIndexCQMetadata(cq: Array[Byte]): DecodedIndexValue = {
+    val sf = decodeIndexCQMetadataToSf(cq)
+    val id = sf.getID
+    val geom = sf.getDefaultGeometry.asInstanceOf[Geometry]
+    val dtg = Option(sf.getAttribute(rasterSftDtgName).asInstanceOf[Date])
+    DecodedIndexValue(id, geom, dtg, null)
+  }
+}
+
+class RasterEntryEncoder extends Logging {
+
+  def encode(raster: Raster, visibility: String = ""): KeyValuePair = {
+
+    logger.trace(s"encoding raster: $raster")
+    val vis = new Text(visibility)
+    val key = new Key(getRow(raster), getCF(raster), getCQ(raster), vis)
+    val encodedRaster = encodeValue(raster)
+
+    (key, encodedRaster)
+  }
+
+  private def getRow(ras: Raster) = {
+    val resEncoder = DoubleTextFormatter(ras.resolution)
+    val geohash = ras.minimumBoundingGeoHash.map(_.hash).getOrElse("")
+    new Text(s"${resEncoder.fmtdStr}~$geohash")
+  }
+
+  //TODO: WCS: add band value to Raster and insert it into the CF here
+  // GEOMESA-561
+  private def getCF(raster: Raster): Text = new Text("")
+  
+  private def getCQ(raster: Raster): Text = {
+    new Text(RasterEntry.encodeIndexCQMetadata(raster.id, raster.metadata.geom, Option(raster.time.toDate)))
+  }
+
+  private def encodeValue(raster: Raster): Value =
+    new Value(raster.serializedChunk)
+
+}
+
+object RasterEntryDecoder {
+  def rasterImageDeserialize(imageBytes: Array[Byte]): RenderedImage = {
+    val in: ObjectInputStream = new ObjectInputStream(new ByteArrayInputStream(imageBytes))
+    var read: RenderedImage = null
+    try {
+      read = in.readObject().asInstanceOf[RenderedImage]
+    } finally {
+      in.close()
+    }
+    read
+  }
+}
+
+import org.locationtech.geomesa.raster.index.RasterEntryDecoder._
+
+class RasterEntryDecoder() {
+  // this should not really need any parameters for the case class, it should simply
+  // construct a Raster from a Key (don't we need the value for this....)
+  // maybe this is not needed at all?
```

Comment:
It is needed, but after looking at this class and the RasterEntryEncoder plus the schema could be re factored to a singleton but I think I will wait till a later commit to make this change